Fix #2514: Add Fixed footer for explorations

joshuacano opened the pull request:
Adding a fixed footer for explorations, Im 90% finished but figured it would be best to get a PR in this for now. I have added a human_readable_contributors_summary item to the get_displayable_exp_summary_dicts in summary_services.py. I will need translation for two keys in i18n, I've already added one for spanish. 

I18N_FOOTER_AUTHOR_PROFILE and I18N_PLAYER_SHARE_EXPLORATION_SIMPLE

Thanks in advance for reviewing this, I'm sure there will be some changes to be made.

seanlip commented on core/templates/dev/head/pages/exploration_player/ExplorationFooterDirective.js:
```diff
@@ -0,0 +1,54 @@
+        $scope.contributorNames = [];
+
+        ExplorationSummaryBackendApiService
+          .loadPublicAndPrivateExplorationSummaries(
```
Good idea to use this method!

seanlip commented on core/templates/dev/head/pages/exploration_player/ExplorationFooterDirective.js:
```diff
@@ -0,0 +1,54 @@
+        ExplorationSummaryBackendApiService
+          .loadPublicAndPrivateExplorationSummaries(
+                [$scope.explorationId]).then(function(summaries) {
+              var summaryBackendObject = null;
```
Replace this and the following with:
```
if (summaries.length > 0) {
  $scope.contributorNames = Object.keys(summaries[0].human_readable_contributors_summary).sort(...);
}
```

where .sort(...) is the logic for sorting the contributors by number of commits, similar to pages/exploration_player/LearnerViewBreadcrumb.js.

joshuacano replied:
Gotcha, I had originally had this. But got it confused in my changes. This has been done.

seanlip commented on core/templates/dev/head/pages/exploration_player/exploration_footer_directive.html:
```diff
@@ -0,0 +1,30 @@
+<script type="text/ng-template" id="components/explorationFooter">
+  <div class="footer navbar-fixed-bottom oppia-exploration-footer">
+    <div class="row">
+        <div class="col-sm-7">
```
Deindent this and the following lines by 2.

seanlip commented on core/templates/dev/head/pages/exploration_player/exploration_footer_directive.html:
```diff
@@ -0,0 +1,30 @@
+  <div class="footer navbar-fixed-bottom oppia-exploration-footer">
+    <div class="row">
+        <div class="col-sm-7">
+          <img ng-src="<[getStaticImageUrl('/general/apple.svg')]>" itemprop="image"
```
Don't need itemprop here. (I think that's just used for page snippets, but this image doesn't represent the content of the page.)

seanlip commented on core/templates/dev/head/pages/exploration_player/exploration_footer_directive.html:
```diff
@@ -0,0 +1,30 @@
+    <div class="row">
+        <div class="col-sm-7">
+          <img ng-src="<[getStaticImageUrl('/general/apple.svg')]>" itemprop="image"
+             style="height: 60px; width: 60px; float:left; margin: -25px 0px 0px 10px;"/>
```
Drop "/" on "/>". Pull the style out into a class and place the class at the bottom of this directive. See examples in other files.

joshuacano replied:
This has been done.

seanlip commented on core/templates/dev/head/pages/exploration_player/exploration_footer_directive.html:
```diff
@@ -0,0 +1,30 @@
+          <img ng-src="<[getStaticImageUrl('/general/apple.svg')]>" itemprop="image"
+             style="height: 60px; width: 60px; float:left; margin: -25px 0px 0px 10px;"/>
+          <div class="dropup">
+            <a id="dropdownMenu2" class="dropdown-toggle" data-toggle="dropdown" style="color:#fff" 
```
Pull styles out into classes. Use a more semantic id, e.g. "authorProfilesDropdownMenu".

joshuacano replied:
have done author-profile-dropdown-menu

seanlip commented on core/templates/dev/head/pages/exploration_player/exploration_footer_directive.html:
```diff
@@ -0,0 +1,30 @@
+             style="height: 60px; width: 60px; float:left; margin: -25px 0px 0px 10px;"/>
+          <div class="dropup">
+            <a id="dropdownMenu2" class="dropdown-toggle" data-toggle="dropdown" style="color:#fff" 
+               ng-if="contributorNames.length>0">
```
Add spaces around ">".

joshuacano replied:
Done.

seanlip commented on core/templates/dev/head/pages/exploration_player/exploration_footer_directive.html:
```diff
@@ -0,0 +1,30 @@
+               ng-if="contributorNames.length>0">
+              <h4 translate="I18N_FOOTER_AUTHOR_PROFILE"></h4>
+            </a>
+            <ul class="dropdown-menu" aria-labelledby="dropdownMenu2" role="menu" style="margin-left: 60px; float: left;">
```
Here and below: pull styles out into classes.

joshuacano replied:
done.

seanlip commented on core/templates/dev/head/pages/exploration_player/exploration_footer_directive.html:
```diff
@@ -0,0 +1,30 @@
+          </div>
+        </div>
+        <div class="col-sm-5">
+        <h4 translate="I18N_PLAYER_SHARE_EXPLORATION_SIMPLE" style="float:left; margin-left:30px; padding-right:10px;text-transform: uppercase;"></h4>
```
This should be indented by 2 from the previous line.

joshuacano replied:
Done.

seanlip commented on core/templates/dev/head/pages/exploration_player/exploration_footer_directive.html:
```diff
@@ -0,0 +1,30 @@
+        </div>
+    </div>
+  </div>
+</script>
```
Add newline at EOF.

joshuacano replied:
Done

seanlip commented on core/templates/dev/head/pages/exploration_player/exploration_footer_directive.html:
```diff
@@ -0,0 +1,30 @@
+        <div class="col-sm-5">
+        <h4 translate="I18N_PLAYER_SHARE_EXPLORATION_SIMPLE" style="float:left; margin-left:30px; padding-right:10px;text-transform: uppercase;"></h4>
+          <sharing-links ng-if="explorationStatus !== 'private'"
+                        ng-click="$event.stopPropagation()"
```
Here and below: add single space to align with previous line.

joshuacano replied:
Done.

seanlip commented on core/templates/dev/head/pages/exploration_player/exploration_footer_directive.html:
```diff
@@ -0,0 +1,30 @@
+        <h4 translate="I18N_PLAYER_SHARE_EXPLORATION_SIMPLE" style="float:left; margin-left:30px; padding-right:10px;text-transform: uppercase;"></h4>
+          <sharing-links ng-if="explorationStatus !== 'private'"
+                        ng-click="$event.stopPropagation()"
+                        twitter-text="DEFAULT_TWITTER_SHARE_MESSAGE_PLAYER"
```
Did you actually test these links? The Twitter one doesn't work for me.

joshuacano replied:
Have corrected for directive. (Never tested once I changed to directive)

seanlip commented on core/templates/dev/head/pages/exploration_player/exploration_player.html:
```diff
@@ -78,6 +78,9 @@
   {% endif %}
+  {% if nav_mode in ['explore'] %}
+    <exploration-footer></exploration-footer>
+            {% endif %}
```
Indentation is wrong.

seanlip replied:
I think the thing to do here might actually be to leave this file alone, and in exploration_player.html, do:

```
{% block footer %}
  <exploration-footer></exploration-footer>
{% endblock footer %}
```

Try it and see? The idea is to replace the footer block in base.html with something custom.

joshuacano replied:
Fixed.

joshuacano replied:
This seems to have worked! Will you double check where I put this is what you were thinking?

seanlip commented on core/templates/dev/head/services/explorationServices.js:
```diff
@@ -19,6 +19,7 @@
 
 // A service that provides a number of utility functions useful to both the
 // editor and player.
+
```
Revert changes to this file.

joshuacano replied:
Done.

seanlip commented on extensions/interactions/EndExploration/EndExploration.js:
```diff
@@ -57,7 +57,7 @@ oppia.directive('oppiaInteractiveEndExploration', [function() {
           // invalid.
           var explorationId = explorationContextService.getExplorationId();
-          $http.get('/explorationsummarieshandler/data', {
+          $http.get(EXPLORATION_SUMMARY_DATA_URL_TEMPLATE, {
```
Optional: Maybe worth changing this to use ExplorationSummaryBackendApiService, too?

joshuacano replied:
Will correct, need to test. In The Meantime, I'll post all of the fixes for your review.

seanlip commented on core/templates/dev/head/pages/exploration_player/LearnerViewBreadcrumb.js:
```diff
@@ -27,7 +27,7 @@ oppia.controller('LearnerViewBreadcrumb', [
         openInformationCardModal();
       } else {
-        $http.get('/explorationsummarieshandler/data', {
+        $http.get(EXPLORATION_SUMMARY_DATA_URL_TEMPLATE, {
```
Optional: Maybe worth changing this to use ExplorationSummaryBackendApiService, too?

joshuacano replied:
Will correct, need to test. In The Meantime, I'll post all of the fixes for your review.

seanlip commented:
Also, /cc-ing @prateekkonduru, @anggorodewanto, @MinhHuong -- would it be possible to get translations for the following strings into Hindi, Indonesian and Vietnamese, please?

- AUTHOR PROFILE
- AUTHOR PROFILES
- Share this exploration

Thanks!

anggorodewanto commented on assets/i18n/id.json:
```diff
@@ -38,6 +38,7 @@
   "I18N_ERROR_PAGE_TITLE_500": "Error 500 - Oppia",
   "I18N_FOOTER_ABOUT": "Tentang",
   "I18N_FOOTER_ABOUT_ALL_CAPS": "TENTANG OPPIA",
+  "I18N_FOOTER_AUTHOR_PROFILE": "",
```
both singular and plural : "Profil penulis"

joshuacano replied:
Thanks! Done.

anggorodewanto commented on assets/i18n/id.json:
```diff
@@ -192,6 +193,7 @@
   "I18N_PLAYER_RETURN_TO_EDITOR": "Kembali ke editor",
   "I18N_PLAYER_RETURN_TO_LIBRARY": "Kembali ke perpustakaan",
   "I18N_PLAYER_SHARE_EXPLORATION": "Nikmati eksplorasi ini? Berbagi dengan teman anda!",
+  "I18N_PLAYER_SHARE_EXPLORATION_SIMPLE": "",
```
"Bagikan eksplorasi ini"

seanlip commented:
Thanks @anggorodewanto!

joshuacano commented:
Great feedback Sean! Thanks, I was finishing up the last bits of this, and I think I addressed the items in the general comments. Will begin addressing the rest of these review comments.

joshuacano commented on core/templates/dev/head/css/oppia.css:
```diff
@@ -671,16 +671,47 @@ textarea {
   font-size: 13px;
-  line-height: 20px;
-  position: absolute;
+  line-height: 10px;
```
Got it, this change was a result of an earlier implementation. Have changed back.

joshuacano commented on core/templates/dev/head/css/oppia.css:
```diff
@@ -671,16 +671,47 @@ textarea {
   width: 100%;
+  margin-top: 30px;
+  height: 42px;
+  font-family: "Capriola", "Roboto", Arial, sans-serif;
```
Got it this has been done.

joshuacano commented on core/templates/dev/head/css/oppia.css:
```diff
@@ -671,16 +671,47 @@ textarea {
   margin-left: auto;
   margin-right: auto;
-  padding: 30px 80px;
+  padding: 3px 80px;
```
No was not, have undone.

joshuacano commented on core/templates/dev/head/css/oppia.css:
```diff
@@ -671,16 +671,47 @@ textarea {
+  color: #fff;
+}
+
+.oppia-sharing-footer {
```
Got it, have removed this rule actually.

joshuacano commented on core/templates/dev/head/css/oppia.css:
```diff
@@ -671,16 +671,47 @@ textarea {
+  margin: -25px 0px 0px 10px;
+}
+
+.oppia-footer-sharing-links ul {
```
Done.

joshuacano commented on core/templates/dev/head/css/oppia.css:
```diff
@@ -671,16 +671,47 @@ textarea {
+    float: left;
+}
+
+.oppia-footer-sharing-links li {
```
Done.

joshuacano replied:
Done and moved to footer_directive

joshuacano commented on core/templates/dev/head/pages/base.html:
```diff
@@ -191,6 +191,7 @@
       {% endif %}
     {% endif %}
 
+
```
Done.

joshuacano commented on core/templates/dev/head/pages/exploration_player/ExplorationFooterDirective.js:
```diff
@@ -0,0 +1,54 @@
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for showing search results.
```
Wow, forgot to change this text to point to Footer. this has been done.

MinhHuong commented:
@Hi @seanlip , I made changes to file assets/i18n/vi.json (adding the new translations). Currently fixing some issues before pushing to git :)

seanlip commented:
Ah -- @MinhHuong, could you just write the translations in this comment thread, please? Then @joshuacano can incorporate them. I think that would be better, so that all the new translations go in at the same time.

Thanks!

MinhHuong commented:
Okie, the translation for those 3 words in Vietnamese :

• AUTHOR_PROFILE : Hồ sơ tác giả
• AUTHOR_PROFILES : Hồ sơ tác giả
• Share this exploration : Chia sẻ cuộc hành trình này

joshuacano commented on assets/i18n/qqq.json:
```diff
@@ -192,6 +193,7 @@
   "I18N_PLAYER_RETURN_TO_EDITOR": "Text displayed when the user is playing an exploration. - Text shown inside a button when the user finishes playing an exploration. When the user clicks the button, the editor page is loaded.",
   "I18N_PLAYER_RETURN_TO_LIBRARY": "Text displayed when the user is playing an exploration. - Text shown inside a button when the user finishes playing an exploration. When the user clicks the button, the library page is loaded.",
   "I18N_PLAYER_SHARE_EXPLORATION": "Text displayed when the user is playing an exploration. - When the user has completed the exploration, this text is shown above the social media buttons used to share an exploration.",
+  "I18N_PLAYER_SHARE_EXPLORATION_SIMPLE": "Text displayed when the user is playing an exploration. - At all times during an exploration, this text is shown in the lower left hand side of the screen next to social media buttons used to share an exploration.",
```
Duh. Thanks, this has been done.

joshuacano commented:
Thanks @MinhHuong!

joshuacano commented:
@seanlip I have addressed the all the comments in your review, Except the fact that the footer is appearing in the embed path.  I'll be making one more commit for that tomorrow. In addition to this, I changed the footer behavior just a bit, by adding a hover to make it more inline with the original mock. (Though it is still not perfectly in sync with the original mockup). Let me know what you think. I can easily change back.

seanlip commented on assets/i18n/en.json:
```diff
@@ -38,6 +38,7 @@
   "I18N_ERROR_PAGE_TITLE_500": "Error 500 - Oppia",
   "I18N_FOOTER_ABOUT": "About",
   "I18N_FOOTER_ABOUT_ALL_CAPS": "ABOUT OPPIA",
+  "I18N_FOOTER_AUTHOR_PROFILE": "AUTHOR PROFILES",
```
I18N_FOOTER_AUTHOR_PROFILES, here and elsewhere.

joshuacano replied:
Done

seanlip commented on assets/i18n/hi.json:
```diff
@@ -38,6 +38,7 @@
   "I18N_ERROR_PAGE_TITLE_500": "Error 500 - Oppia",
   "I18N_FOOTER_ABOUT": "हमारे बारे में",
   "I18N_FOOTER_ABOUT_ALL_CAPS": "OPPIA के बारे में",
+  "I18N_FOOTER_AUTHOR_PROFILE": "",
```
@souravbadami or @526avijitgupta -- do you think you could help with these translations, please? Thanks!

seanlip replied:
@joshuacano, I got some translations (thanks to @raju249!) Here they are:

AUTHOR PROFILES: लेखक प्रोफाइल
Share this exploration: शेयर दीस एक्सप्लोरेशन

joshuacano replied:
Thanks! Done.

seanlip commented on assets/i18n/pt-br.json:
```diff
@@ -192,6 +192,7 @@
   "I18N_PLAYER_RETURN_TO_EDITOR": "Voltar ao editor",
   "I18N_PLAYER_RETURN_TO_LIBRARY": "Voltar à biblioteca",
   "I18N_PLAYER_SHARE_EXPLORATION": "Gostou desta exploração? Compartilhe com seus amigos!",
+  "I18N_PLAYER_SHARE_THIS_EXPLORATION" : "",
```
I don't see I18N_FOOTER_AUTHOR_PROFILES here...?

joshuacano replied:
Added. Wonder why this wasnt caught in the i18n checks of testing? Just curious.

seanlip replied:
Oh, yeah. I remember wondering that too. Not sure...

seanlip commented on assets/i18n/pt.json:
```diff
@@ -38,6 +38,7 @@
   "I18N_ERROR_PAGE_TITLE_500": "Error 500 - Oppia",
   "I18N_FOOTER_ABOUT": "Sobre",
   "I18N_FOOTER_ABOUT_ALL_CAPS": "SOBRE O OPPIA",
+  "I18N_FOOTER_AUTHOR_PROFILE": "",
```
@joshuacano -- need to change the translation here?

seanlip commented on assets/i18n/pt.json:
```diff
@@ -192,6 +193,7 @@
   "I18N_PLAYER_RETURN_TO_EDITOR": "Regressar ao editor",
   "I18N_PLAYER_RETURN_TO_LIBRARY": "Regressar à biblioteca",
   "I18N_PLAYER_SHARE_EXPLORATION": "Gostaste desta exploração? Partilha-a com os teus amigos!",
+  "I18N_PLAYER_SHARE_THIS_EXPLORATION": "",
```
Partilha esta exploração

joshuacano replied:
Done.

seanlip commented on core/templates/dev/head/css/oppia.css:
```diff
@@ -671,15 +671,16 @@ textarea {
   color: #fff;
   font-size: 13px;
-  line-height: 20px;
+  line-height: 20px
```
Missing semicolon at end.

joshuacano replied:
Done

seanlip commented on core/templates/dev/head/pages/exploration_player/exploration_footer_directive.html:
```diff
@@ -0,0 +1,134 @@
+                <h4 translate="I18N_FOOTER_AUTHOR_PROFILE" class="author-profile-text"></h4>
+              </a>
+              <ul class="author-profile-dropdown-menu dropdown-menu" aria-labelledby="dropdownMenu2" role="menu">
+                <li ng-repeat="name in contributorNames | limitTo: 5" style=>
```
Clicking on the li should open /profile/{{author_username}} in a new tab.

seanlip replied:
Why the "style=" at the end?

joshuacano replied:
Accident! removed

joshuacano replied:
Done. I believe Will you check this for me? Is the contributorName the same as the author_username?

seanlip replied:
Yep, I believe so. If you click on the link, you should visit /profile/{{name}} (assuming this is in the context of "name in contributorNames").

joshuacano replied:
Thanks!

seanlip commented on core/templates/dev/head/pages/exploration_player/exploration_footer_directive.html:
```diff
@@ -0,0 +1,134 @@
+          <li class="hover-link"
+              ng-mouseenter="hovering=true"
+              ng-mouseleave="hovering=false">
+            <div class="dropup" ng-class="{'open':hovering}">
```
nit: here and below: add space after colon

joshuacano replied:
Done

seanlip commented on core/templates/dev/head/pages/exploration_player/exploration_footer_directive.html:
```diff
@@ -0,0 +1,134 @@
+            <img ng-src="<[getStaticImageUrl('/general/apple.svg')]>" class="exploration-footer-img">
+          </li>
+          <li class="hover-link"
+              ng-mouseenter="hovering=true"
```
nit: here and below -- add spaces around '='

joshuacano replied:
Done

seanlip commented on core/templates/dev/head/pages/exploration_player/exploration_footer_directive.html:
```diff
@@ -0,0 +1,134 @@
+
+<style>
+
+.oppia-exploration-footer {
```
Indent all this (and the following) by 2.

joshuacano replied:
Done

seanlip commented on core/templates/dev/head/pages/exploration_player/exploration_footer_directive.html:
```diff
@@ -0,0 +1,134 @@
+  width: 100%;
+}
+
+.exploration-footer-img {
```
Prefix every selector with .oppia-exploration-footer, so that the styles in this file apply only to the footer and don't proliferate all over the application.

joshuacano replied:
Done.

seanlip commented on core/templates/dev/head/pages/exploration_player/exploration_footer_directive.html:
```diff
@@ -0,0 +1,134 @@
+.exploration-footer-img {
+  height: 60px;
+  width: 60px;
+  margin: -25px 0px 0px 10px;
```
Alphabetize; 'margin' should come between 'height' and 'width'. There are a number of places below where this sort of thing also needs to be fixed.

joshuacano replied:
Gotcha, this and all other places in the file have been fixed.

seanlip commented on core/templates/dev/head/pages/exploration_player/exploration_footer_directive.html:
```diff
@@ -0,0 +1,134 @@
+}
+
+.oppia-exploration-footer .author-profile .hovered-text h4 {
+  color: #094142 !important;
```
Is it possible to use a more specific selector/class instead? We tend to eschew `!important` since it overrides the usual order.

joshuacano replied:
Totally, It felt wrong using this in the first place. Have revised to handle a more specific class.

seanlip commented on core/templates/dev/head/pages/exploration_player/exploration_footer_directive.html:
```diff
@@ -0,0 +1,134 @@
+  float: left;
+  border-radius: 0;
+  list-style-type: none; 
+
```
Drop unnecessary newline.

joshuacano replied:
Done

seanlip commented:
Hi @joshuacano, here are the Brazilian translations (thanks, @lawrencelm!):

- AUTHOR PROFILES = PERFIS DE AUTORES
- Share this exploration = Compartilhe essa exploração

seanlip commented:
Hi @joshuacano -- one other UI thing that I noticed while reviewing this: in the bottom right of the exploration player page, there is actually a CC-license indicator, but it is now obscured by the footer and the share buttons. Would you be able to move it higher, please?

Thanks!

joshuacano commented:
@seanlip I have addressed all comments and moved the CC-license up a few pixels, let me know if it looks good. Thanks for the embed help offline, I have placed that functionality in an {% if not iframed %} block. I also noticed an error when displaying more than one contributor, so this should look good with multiple contributors..

seanlip commented:
Hi @joshuacano, I think this looks very nice -- great job! Just a couple of small things.

- When hovering on "Author Profiles", the full profile is not selected, so it looks very strange. See screenshot below. Could this be fixed?

![wholecr](https://cloud.githubusercontent.com/assets/10575562/20230056/a478fefe-a80e-11e6-920f-95ee58235af2.png)

- Please could you move the CC-thing higher by about 6 px? It is currently a bit too close to the share buttons, I think.

Thanks!

seanlip commented on core/templates/dev/head/pages/exploration_player/ExplorationFooterDirective.js:
```diff
@@ -0,0 +1,63 @@
+                  Object.keys(
+                    summaries[0].human_readable_contributors_summary).sort(
+                      function(contributorUsername1, contributorUsername2) {
+                        var commitsOfContributor1 = dict[
```
@joshuacano -- There's reference to a "dict" here, but that dict is not defined, and this results in a frontend error. I noticed this when testing the case of 2 contributors. This needs to be fixed...

To simulate:
1. Log in as A, then create and publish an exploration.
2. In a separate browser or incognito window, sign up a new user, B.
3. A then adds B as a collaborator in the Settings tab of A's exploration.
4. B goes to A's exploration and edits it.
5. Go to /explore/{{exploration_id}} and you should see the error.

Thanks!

joshuacano replied:
I just completely overlooked this, its obvious looking at it now. Have fixed.

seanlip commented:
Also, @joshuacano -- I think your changes broke some e2e tests. Could you please take a look at those and fix the issue? https://travis-ci.org/oppia/oppia/builds/175164451

Thanks!

joshuacano commented:
@seanlip I've resolve the cc-icon issue, the hover link item and the e2e tests. Should be good to go. I just wanted to double check one thing, the e2e_tests were failing because the global EXPLORATION_SUMMARY_DATA_URL_TEMPLATE was not injected into the directive, this needs to be explicit for each directive that is using it, correct? Just wanted to verify. Thanks for your help!

seanlip commented on core/templates/dev/head/pages/exploration_player/ExplorationFooterDirective.js:
```diff
@@ -45,13 +45,14 @@ oppia.directive('explorationFooter', [function() {
                 [$scope.explorationId]).then(function(summaries) {
               var summaryBackendObject = null;
               if (summaries.length > 0) {
+                contributorSummary = (
```
Missing `var`

seanlip commented:
With regards to EXPLORATION_SUMMARY_DATA_URL_TEMPLATE -- yes, that's correct! Any external "thing" that the directive uses must be injected explicitly.

seanlip commented:
Hi @joshuacano -- apart from the tiny code comment I left, this looks really good!

I played through the UI and I only have one comment. When hovering over the author profiles, could you make the entire "row" clickable, similar to the About dropdown on /splash? I.e., if you have a short-ish username, hovering over the white space to the right of it should cause the whole row to highlight more visibly, and clicking anywhere on this row (not just the username) would open the profile link. I think that would be much better for visibility.

Other than that, I agree that this is ready to go in. I couldn't find any more issues.

Thanks!

joshuacano commented:
This has been done! Thanks @seanlip!
